### PR TITLE
[CC40] Add LTR/RTL Language Support to Templates / STYLE UPDATES BASED ON V7 WIREFRAMES

### PR DESCRIPTION
**Pull request by Audiosutras**
## Description

https://caktus.atlassian.net/browse/CC-40
Wireframes:OLD: https://www.figma.com/proto/k2XxlRS0JpII6NXDd3K1zc/Mockup?node-id=395%3A180&scaling=min-zoomNEW: https://www.figma.com/proto/k2XxlRS0JpII6NXDd3K1zc/Mockup?node-id=986%3A0&viewport=212%2C45%2C0.07671261578798294&scaling=min-zoom

1. RTL and LTR language support has been added
2. Added Subscriber newsletter Div

Noticed differences:
1. Creative Commons has been removed from the first line in the sidebar menu on licenses/license
2. Update Related-Links section which also now supports rtl/ltr languages.

## Tests

Checking `Right to Left` and `Left to Right` Language Support: 
In `cc_licenses.templates.base.html` on `line 18` see `{% with lang_dir="ltr" %}`. `lang_dir` will be supplied by the `licenses.views`. For checking language accessibility and support for the templates simply toggle `lang_dir` between `ltr` and `rtl`. Toggle `lang_dir` in `base.html` will effect all child templates (those being `licenses_detail`, `deed_detail`, and `sampling_deed_detail`).

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dpoirier — COMMENTED** [review]
"ltr" or "rtl" should be set based on the current language. For example, we could look at https://docs.djangoproject.com/en/3.1/topics/i18n/translation/#get-current-language-bidi.

It'd be nice if we could reduce the number of ``if`` statements in the templates. E.g. if instead of ``{% if lang_dir == 'ltr' %}left{% else %}right{% endif %}``, we could write something like ``{{ start }}`` or ``{% start %}``, so the result looks like ``padding-{% start %}-normal`` instead of ``padding-{% if lang_dir == 'ltr' %}left{% else %}right{% endif %}-normal``. Which is still kind of ugly, but at least better. It's too bad Bulma and Vocabulary don't directly support adjusting padding independent of language direction using something like ``padding-inline-start`` under the covers.

When I force "rtl" by editing the place lang_dir is set, and go to http://localhost:8000/licenses/deed/, it looks like a little more space is needed after some of the icons.

image

**Comment by Audiosutras** [comment]
Pushed comment for adding `templatetags` that will more clearly manage `rtl`/`ltr` in templates. For icon padding, I need to check with the CC-Vocabulary (@zackkrida) and see if vocabulary offers responsive classes or if there is a particular device breakpoint we should be focusing on.

**Comment by zackkrida** [comment]
> if vocabulary offers responsive classes

Not currently, no. There are two ways you can do this:

1. Duplicate code and show/hide elements based on viewport. Use _these_ helper classes: https://bulma.io/documentation/helpers/visibility-helpers/#hide, For example, you might have `<h1 class="fake-huge-text is-hidden-touch">Page Title</h1>` and `<h1 class="fake-smaller-text is-hidden-desktop">Page Title</h1>` on the page at the same time
2. Write custom styles with media queries (in css) or use bulma's media helpers in sass:

```scss
@import 'bulma/sass/utilities/_all.sass';

.fake-component {
	display: block;

	@include touch {
        display: flex;
    }
}
```

File: licenses/templatetags/lang_pad_horizontal.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+@register.simple_tag
+def lang_pad_horizontal(is_rtl, is_reverse):
+    """Adjust padding left or right based on language direction
+
+    Parameter(s):
+        - is_rtl - (boolean) whether or not the langauge of the
+                 - of the template is a right to left language
+    """
```
Comment (dpoirier) [inline licenses/templatetags/lang_pad_horizontal.py:13]: Can you add an explanation of what `is_reverse` is for and when I should pass False or True?

Or maybe we could have two tags. (I may be remembering this from another project a few years ago.) Something like

    @register.simple_tag(takes_context=True)
    def start(context):
        return "right" if context["is_rtl"] else "left"

and another one `end` that returns the opposite.  Usage might look like `class="padding-{% start %}-normal"` if you wanted normal padding on the left normally and on the right if we're in RTL mode.

**Comment by Audiosutras** [comment]
Currently waiting on feedback about rtl support for the `header` and `breadcrumb` sections from the vocabulary team. @zackkrida @dpoirier

**Comment by zackkrida** [comment]
@Audiosutras here are some styles that will fix the dropdown arrows and menu item placement when rtl (screenshot is still using english, but you get the idea):

Screen Shot 2020-09-09 at 6 50 15 AM

```css
html[dir="rtl"] .has-dropdown .navbar-link {
    flex-direction: row-reverse;
}

html[dir="rtl"] .navbar-end {
    margin-right: auto;
    margin-left: 0;
    justify-content: flex-start;
}
```

**Comment by Audiosutras** [comment]
> @Audiosutras here are some styles that will fix the dropdown arrows and menu item placement when rtl (screenshot is still using english, but you get the idea):
> 
> Screen Shot 2020-09-09 at 6 50 15 AM
> 
> ```css
> html[dir="rtl"] .has-dropdown .navbar-link {
>     flex-direction: row-reverse;
> }
> 
> html[dir="rtl"] .navbar-end {
>     margin-right: auto;
>     margin-left: 0;
>     justify-content: flex-start;
> }
> ```

@zackkrida thank you.

File: licenses/templatetags/lang_pad_horizontal.py (hunk 1 above)
Comment (Audiosutras) [reply licenses/templatetags/lang_pad_horizontal.py]: done

Comment (Audiosutras) [reply licenses/templatetags/lang_pad_horizontal.py]: Originally, you asked me to include:
```
{% get_language_info for 'en' as lang %}
{% with is_rtl=lang.bidi %}
```
You said we didn't want the view to handle language direction. Just for it to supply the language. The above example is using context. Should the above code be removed as well?

**Review by dpoirier — COMMENTED** [review]
I think I'd do something like

```
<!DOCTYPE html>
{% get_current_language as LANGUAGE_CODE %}{# View will have set current language #}
{% get_language_info for LANGUAGE_CODE as lang %}
<html lang="{{ lang.name }}" class="no-js" dir="{% if lang.bidi %}rtl{% else %}ltr{% endif %}">
```

and

```
@register.simple_tag
def start():
    """Returns 'left' if ltr language or 'right' if rtl language """
    return "right" if get_language_bidi() else "left"
```

so no need for is_rtl variable.

The lang_dir tag isn't being used anymore and can be removed. The start and end tags can be put in the same file.

**Comment by Audiosutras** [comment]
> get_language_bidi()

Ok, I'm going to copy what you have here and commit it.

Its much cleaner.

**Comment by Audiosutras** [comment]
> I think I'd do something like
> 
> ```
> <!DOCTYPE html>
> {% get_current_language as LANGUAGE_CODE %}{# View will have set current language #}
> {% get_language_info for LANGUAGE_CODE as lang %}
> <html lang="{{ lang.name }}" class="no-js" dir="{% if lang.bidi %}rtl{% else %}ltr{% endif %}">
> ```
> 
> and
> 
> ```
> @register.simple_tag
> def start():
>     """Returns 'left' if ltr language or 'right' if rtl language """
>     return "right" if get_language_bidi() else "left"
> ```
> 
> so no need for is_rtl variable.
> 
> The lang_dir tag isn't being used anymore and can be removed. The start and end tags can be put in the same file.

If possible, can you add an additional commit with these changes.

**Review by dpoirier — APPROVED** [review]

**Pull request merged**